### PR TITLE
feat: Add `successRedirectUrl` to auth options

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**

Add `successRedirectUrl` to provider options. The option can be used to configure a URL for the callback to redirect to.

For example, you could have the following flow:
- `POST /auth/admin/google`
- Redirect to Google auth
- `POST /auth/admin/google/callback`
  - Validate callback
- Redirect to admin with token in URL e.g. `http://localhost:5173/login?access_token=...`
- Decode token client-side
- `POST /admin/users` with auth user from token

If no `successRedirectUrl` is configured, we respond with the token from the callback.

### CODE REVIEW

File: packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts [hunk 1]
```diff
@@ -20,18 +20,26 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
...
+    const user = { ...authUser }
+    const token = jwt.sign(user, jwt_secret)
+
+    if (successRedirectUrl) {
+      const url = new URL(successRedirectUrl!)
+      url.searchParams.append("auth_token", token)
+
+      const redirectUrl = `${url}`
+
+      return res.redirect(redirectUrl.toString())
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts]:
```suggestion
      return res.redirect(redirectUrl)
```
nit

**Review by carlos-r-l-rodrigues — APPROVED** [review]

File: packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts [hunk 2]
```diff
@@ -20,18 +20,26 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
...
-    return
-  }
+  const { success, error, authUser, successRedirectUrl } = authResult
 
   if (success) {
     const { jwt_secret } = req.scope.resolve("configModule").projectConfig
-    const token = jwt.sign(authUser, jwt_secret)
-    return res.status(200).json({ token })
+
+    const user = { ...authUser }
```
Comment (sradevski) [inline packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts]: Why do we need to destructure the authUser here?

File: packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts (hunk 1 above)
Comment (sradevski) [reply packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts]: Or just do `url.toString()` which I think is better.

**Review by sradevski — APPROVED** [review]

File: packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts (hunk 2 above)
Comment (olivermrbl) [reply packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts]: The payload to `jwt.sign` needs to be a plain object and it was being returned as an entity object here. 

I revisited the approach and decided to serialize the auth user response at the service level in 8e8681a50be0d70a9dfdcddafd0238c0a1cc1ac3, so we don't need this dirty destructuring.

Comment (sradevski) [reply packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts]: @olivermrbl ah I see, yeah that sound better indeed

**Review by sradevski — APPROVED** [review]

**Pull request merged**